src/CocktailDetails/CocktailDetails.tsx

```diff
@@ -1,66 +1,125 @@
-import React, { useState, useEffect } from 'react';
-import { getCocktailDetails } from '../apiCalls';
-import { Cocktail } from '../Definitions/RandomCocktail'
-import './CocktailDetails.scss';
+import React, { useState, useEffect } from "react";
+import { getCocktailDetails } from "../apiCalls";
+import { Cocktail } from "../Definitions/RandomCocktail";
+import "./CocktailDetails.scss";
+
 export interface CocktailDetailsProps {
-	id: string;
+  id: string;
+  toggleFavorites: (id: string) => any;
+  favCocktails: string[];
 }
 
-const CocktailDetails: React.FC<CocktailDetailsProps> = (props) => {
-	const [cocktailInfo, setCocktailInfo] = useState<Cocktail>({idDrink: '', strDrink: '', strInstructions: '', strDrinkThumb: ''})
-	const [error, setError] = useState('');
+const CocktailDetails: React.FC<CocktailDetailsProps> = ({ id, toggleFavorites, favCocktails }) => {
+  const [cocktailInfo, setCocktailInfo] = useState<Cocktail>({
+    idDrink: "",
+    strDrink: "",
+    strInstructions: "",
+    strDrinkThumb: "",
+  });
+  const [error, setError] = useState("");
+  const [isFavorite, setIsFavorite] = useState<boolean>(
```

Comment:
state for rendering appropriate button based on a users favorites array

Reply:
😎  I like it